diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +123,161 @@ class LineLength : Rule("line-length") {
     }
 
+    private fun fixError(wrongNode: ASTNode, configuration: LineLengthConfiguration) {
+        var parent = wrongNode
+        do {
+            when (parent.elementType) {
+                FUN -> {
+                    if (!parent.hasChildOfType(BLOCK))
+                        parent.addChild(PsiWhiteSpaceImpl("\n"), parent.findChildByType(EQ)!!.treeNext)
+                    return
+                }
+                CONDITION -> {
+                    fixCondition(parent, configuration)
+                    return
+                }
+                PROPERTY -> {
+                    fixProperty(parent, configuration)
+                    return
+                }
+                EOL_COMMENT -> {
+                    fixComment(wrongNode, configuration)
+                    return
+                }
+                else -> parent = parent.treeParent
+            }
+        } while (parent.treeParent != null)
+    }
+
+    private fun fixComment(wrongNode: ASTNode, configuration: LineLengthConfiguration) {
+        val leftOffset = positionByOffset(wrongNode.startOffset).second
+        val indexLastSpace = wrongNode.text.substring(0, configuration.lineLength.toInt() - leftOffset).lastIndexOf(' ')
+        if (indexLastSpace == -1)
+            return
+        val nodeText = "//${wrongNode.text.substring(indexLastSpace, wrongNode.text.length)}"
+        wrongNode.treeParent.run {
+            addChild(LeafPsiElement(EOL_COMMENT, wrongNode.text.substring(0, indexLastSpace)), wrongNode)
+            addChild(PsiWhiteSpaceImpl("\n"), wrongNode)
+            addChild(LeafPsiElement(EOL_COMMENT, nodeText), wrongNode)
+            removeChild(wrongNode)
+        }
+    }
+
+    private fun fixCondition(wrongNode: ASTNode, configuration: LineLengthConfiguration,
+                             leftInitOffset: Int = -1, isProperty: Boolean = false) {
+        var leftOffset = if (leftInitOffset < 0) {
+            positionByOffset(wrongNode.firstChildNode.startOffset).second
+        } else leftInitOffset
+        val binList = mutableListOf<ASTNode>()
+        if (isProperty)
+            dfsForProperty(wrongNode, binList)
+        else
+            searchBinaryExpression(wrongNode, binList)
+        var binaryText = ""
+        binList.forEachIndexed { index, astNode ->
+            if (index == 0) {
+                binaryText += astNode.treeParent.prevSibling { it.elementType == LPAR}?.text ?: ""
+                binaryText += astNode.prevSibling { it.elementType == WHITE_SPACE}?.text ?: ""
+                binaryText += astNode.text
+                binaryText += astNode.nextSibling { it.elementType == WHITE_SPACE}?.text ?: ""
+            } else {
+                binaryText += astNode.treeParent.prevSibling { it.elementType == LPAR}?.text ?: ""
+                binaryText += astNode.prevSibling { it.elementType == WHITE_SPACE}?.text ?: ""
+                binaryText += astNode.treeParent.findChildByType(OPERATION_REFERENCE)!!.text
```

Comment:
I don't quite get it: if `binList` contains `BINARY_EXPRESSION` nodes, don't they contain operation reference as a child node? That said, why doesn't `astNode.text` contain operator and you need to explicitly add it from another node instead?

Reply:
Oh, I see, operator from binary expression is a child of `astNode`, while you look for `OPERATION_REFERENCE` among it's siblings. But I still don't quite understand, what does this operation reference correspond to? Why is it always not nullable?

Reply:
Yeah, that's because you handle the first node independently, and if there are several nodes, there is an operator between them.